pkg/database/database.go

```diff
@@ -0,0 +1,57 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/kubearchive/kubearchive/pkg/models"
+	_ "github.com/lib/pq"
+)
+
+type DBInterface interface {
+	QueryResources(kind, group, version string) ([]models.Resource, error)
+}
+
+type Database struct {
+	db                *sql.DB
+	resourceTableName string
+}
+
+func NewDatabase() (*Database, error) {
+	dataSource, err := ConnectionStr()
+	if err != nil {
+		return nil, err
+	}
+	db, err := sql.Open("postgres", dataSource)
+	if err != nil {
+		return nil, err
+	}
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+	return &Database{db, "archive"}, nil
+}
+
+func (db *Database) QueryResources(kind, group, version string) ([]models.Resource, error) {
```

Comment:
can you add `ctx context.Context` as the first parameter to this method? This will allow the pass a timeout to the database. This will also make it easier to add otel instrumentation for this later

```suggestion
func (db *Database) QueryResources(ctx context.Context, kind, group, version string) ([]models.Resource, error) {
```

Reply:
updated!